rtos/Mutex.h

```diff
@@ -172,7 +174,7 @@ class Mutex : private mbed::NonCopyable<Mutex> {
 
 private:
 #if MBED_CONF_RTOS_PRESENT
-    void constructor(const char *name = nullptr);
+    void constructor(const char *name = nullptr, bool recursive = true);
```

Comment:
This shouldn't really have default parameter(s). I did this same pattern the other day, and it allowed an error where I forgot to pass that parameter from the outer constructor, and it just defaulted:

     MyClass(int param1, int param2) {
          inner_constructor(param1);
     }

Reply:
We used default parameters in the constructors' signature to save us from creating two extra constructors with and without name plus the recursive arg.
The default parameter in this function is necessary to stay consistent in the order of the parameters.

Reply:
I'm talking about the default parameters of this private function.